### PR TITLE
Remove duplicate attachment reference checking and add image layout check for duplicate references.

### PR DESCRIPTION
**Pull request by asuonpaa**
There was a check that attachment referenced don't have duplicate attachment IDs. Spec says it's allowed as long as they have identical image layouts. The duplicate reference check was removed, and a check for the image layouts was added. There's also some code cleanup to reuse the same renderpass2 related code repeated in several tests.

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -10929,9 +10929,11 @@ static bool AddAttachmentUse(const layer_data *dev_data, RenderPassCreateVersion
     const char *const function_name = use_rp2 ? "vkCreateRenderPass2KHR()" : "vkCreateRenderPass()";
 
     if (uses & new_use) {
-        log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_UNKNOWN_EXT, 0,
-                kVUID_Core_DrawState_InvalidRenderpass, "%s: subpass %u already uses attachment %u as a %s attachment.",
-                function_name, subpass, attachment, StringAttachmentType(new_use));
+        if (attachment_layouts[attachment] != new_layout) {
+            vuid = use_rp2 ? "VUID-VkSubpassDescription2KHR-layout-02528" : "VUID-VkSubpassDescription-layout-02519";
+            log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_UNKNOWN_EXT, 0, vuid,
+                    "%s: subpass %u already uses attachment %u with a different image layout.", function_name, subpass, attachment);
```
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp]: include the two conflicting layouts in the message -- `string_VkImageLayout` to convert them to human readable strings.

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -211,6 +211,17 @@ static VkSamplerCreateInfo SafeSaneSamplerCreateInfo() {
     return sampler_create_info;
 }
 
+// Helper for checking createRenderPass2 support and adding related extensions.
+static bool CheckCreateRenderPass2Support(VkRenderFramework *renderFramework, std::vector<const char *> &device_extension_names) {
+    if (renderFramework->DeviceExtensionSupported(renderFramework->gpu(), nullptr, VK_KHR_CREATE_RENDERPASS_2_EXTENSION_NAME)) {
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: each dependent extension should be checked as well.

File: tests/layer_validation_tests.cpp [hunk 3]
```diff
@@ -5474,22 +5451,9 @@ TEST_F(VkLayerTest, RenderPassCreateAttachmentDescriptionInvalidFinalLayout) {
     }
 
     ASSERT_NO_FATAL_FAILURE(InitFramework(myDbgFunc, m_errorMonitor));
-    PFN_vkCreateRenderPass2KHR vkCreateRenderPass2KHR = nullptr;
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: I like the refactor.  Usually we put these in their own own commit.  If not too hard, that would be the preference.

So one commit would hold the implementation of the utility function and all of the code changes needed to use it.

File: tests/layer_validation_tests.cpp [hunk 4]
```diff
@@ -5402,67 +5392,54 @@ TEST_F(VkLayerTest, RenderPassCreateAttachmentReadOnlyButCleared) {
...
 
-        TestRenderPassCreate(m_errorMonitor, m_device->device(), &rpci, vkCreateRenderPass2KHR,
+        TestRenderPassCreate(m_errorMonitor, m_device->device(), &rpci, rp2Supported,
                              "VUID-VkRenderPassCreateInfo-pAttachments-01567", nullptr);
     }
 }
 
-TEST_F(VkLayerTest, RenderPassCreateAttachmentUsedTwiceColor) {
-    TEST_DESCRIPTION("Attachment is used simultaneously as two color attachments. This is usually unintended.");
+TEST_F(VkLayerTest, RenderPassCreateAttachmentMismatchingLayoutsColor) {
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: This is correct combined with the layer change, as the two are closely coupled, and there is no straightforward, bisectable way to split the test and layer changes in this instance. 

For future reference, _usually_ we would expect test changes in their own commit subsequent to the layer change -- such that layer_validation_tests passes for each commit.

File: tests/layer_validation_tests.cpp (hunk 2 above)
Comment (daveh-lunarg) [reply tests/layer_validation_tests.cpp]: When the primary extension is dependent on the others per the spec, it's existence implies them as well (barring a driver error). There are examples of both in the test code already, so I would say there's no harm in checking, but not a requirement.

Comment (jzulauf-lunarg) [reply tests/layer_validation_tests.cpp]: @daveh-lunarg -- 100% correct on dependencies.  No change is needed here.  

From 30.3 : 
>If an extension is supported (as queried by vkEnumerateInstanceExtensionProperties or vkEnumerateDeviceExtensionProperties), then required extensions of that extension must also be supported for the same instance or physical device.
>
> Any device extension that has an instance extension dependency that is not enabled by vkCreateInstance is considered to be unsupported, hence it must not be returned by vkEnumerateDeviceExtensionProperties for any VkPhysicalDevice child of the instance.

File: layers/core_validation.cpp (hunk 1 above)
Comment (asuonpaa) [reply layers/core_validation.cpp]: Done

File: tests/layer_validation_tests.cpp (hunk 3 above)
Comment (asuonpaa) [reply tests/layer_validation_tests.cpp]: They are now in separate commits.

File: tests/layer_validation_tests.cpp [hunk 5]
```diff
@@ -5420,8 +5420,8 @@ TEST_F(VkLayerTest, RenderPassCreateAttachmentReadOnlyButCleared) {
...
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: @cnorthrop -- here is a test with a name change.  Is this going to affect your internal CI?

Comment (cnorthrop) [reply tests/layer_validation_tests.cpp]: I don't believe that test name is being special cased anywhere.

**Comment by jzulauf-lunarg** [comment]
fixes #578
